.github/workflows/pullrequest.yml

```diff
@@ -1,29 +1,37 @@
-# This workflow will build a package using Maven and then publish it to GitHub packages when a release is created
-# For more information see: https://github.com/actions/setup-java#apache-maven-with-a-settings-path
-
 name: maven-build
 
 on:
-    pull_request:
-      branches:
-        - master
+  pull_request:
+    branches:
+      - master
+    paths-ignore:
+      - '*.md'
```

Comment:
Would the necessary checks be green if there is only a change in the readme file?

Reply:
Sounded like a basic question but is indeed an interesting issue. According to [the GitHub docs](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/collaborating-on-repositories-with-code-quality-features/troubleshooting-required-status-checks#handling-skipped-but-required-checks), if a certain check is required and will be skipped because of some path rules, the check remains in a pending state. The branch `master` is protected and requires the check `build` to succeed. This means that a PR which changes only `.md` files cannot become green.

We should therefore ignore this optimisation of skipping certain paths for the `pullrequest` workflow. The `release` workflow can still ignore certain paths because it does not contain any required checks. I will commit the necessary changes.